quil-rs/src/program/graph.rs

```diff
@@ -516,22 +519,19 @@ fn terminate_working_block(
         return Ok(());
     }
 
-    let block = InstructionBlock::build(working_instructions.to_vec(), terminator, program)?;
+    // This leaves working_instructions and working_label in their respective "empty" states.
+    let block = InstructionBlock::build(std::mem::take(working_instructions), terminator, program)?;
     let label = working_label
-        .clone()
+        .take()
         .unwrap_or_else(|| ScheduledProgram::generate_autoincremented_label(blocks));
 
-    match blocks.insert(label.clone(), block) {
-        Some(_) => Err(ScheduleError {
+    if blocks.insert(label.clone(), block).is_some() {
+        return Err(ScheduleError {
             instruction_index,
             instruction: Instruction::Label(Label(label)),
             variant: ScheduleErrorVariant::DuplicateLabel,
-        }),
-        None => Ok(()),
-    }?;
-
-    working_instructions.drain(..);
-    *working_label = None;
```

Comment:
Simplified the logic of this `match` statement while I was here.

Reply:
You are correct: the logic is slightly different. I called those changes out [here](https://github.com/rigetti/quil-rs/pull/218#discussion_r1224894330) and [here](https://github.com/rigetti/quil-rs/pull/218#discussion_r1224894789). In short, I did not see any attempt to recover from the errors, so it seems safe to empty the values before erroring.

Reply:
I see, reviewing the code more, I don't think this change in logic will introduce any problem. For future changes, maybe just add a note to the docstring that the function _will_ wipe out `working_instructions` and `working_label`.